File: frontend/src/types/route.ts
export interface RouteStep {
  instruction: string;
  distance_meters: number;
  duration_seconds: number;
  maneuver_type: string;
}

export interface RouteData {
  distance_meters: number;
  duration_seconds: number;
  geometry: {
    type: "LineString";
    coordinates: [number, number][];
  };
  steps: RouteStep[];
}

File: backend/app/models/route.py
"""Pydantic models for the Mapbox Directions route response."""

from typing import Any, Dict, List

from pydantic import BaseModel


class RouteStep(BaseModel):
    instruction: str
    distance_meters: float
    duration_seconds: float
    maneuver_type: str  # "depart" | "turn" | "arrive" | ...


class RouteResponse(BaseModel):
    distance_meters: float
    duration_seconds: float
    # GeoJSON LineString — passes through as-is to the frontend
    geometry: Dict[str, Any]
    steps: List[RouteStep]
